### PR TITLE
Added No-Server mode

### PR DESCRIPTION
**Pull request by LTHeaven**
This pull request includes a

- [ ] Bug fix
- [x] New feature
- [ ] Translation

The following changes were made

-  Added No-Server mode (flag -ns). Effect: Only starts the searcher but not the webserver

### CODE REVIEW

**Comment by JZ6** [comment]
Woo pr #1000!

**Comment by JZ6** [comment]
is there a better way of implmentation without using time.sleep?

**Comment by MonkeyLeeT** [comment]
Why not just
`If not args.no_server:  
  app.run()`
?

**Comment by LTHeaven** [comment]
i have to keep the main thread alive thatswhy the while loop.

**Comment by MonkeyLeeT** [comment]
I'm not most familiar with python. Is that how to keep daemon thread alive, by sleep(1000)?

**Comment by LTHeaven** [comment]
the problem is when the main thread ends the search thread also ends. so i make a loop and use time.sleep to reduce the interpreter stress

**Comment by invisiblek** [comment]
what about this? https://gist.github.com/invisiblek/e6f6cec123ec4431ee48475c48d20aa6

**Comment by devinsba** [comment]
@LTHeaven is your plan to share a database on one web server with a bunch of searchers?

**Comment by kingvibs** [comment]
@LTHeaven @invisiblek if that is the case then why not implement use of DHT (distributed hash table) or similar, so nodes can pop in and out and reduce your server loads

**Comment by devinsba** [comment]
@kingvibs That seems like not a good fit for the problem. Could easily be done with a simple change in database type. Ie: switch to not sqlite (mysql, postgres, cassandra, etc.)

File: runserver.py [hunk 1]
```diff
@@ -60,4 +61,8 @@ def start_locator_thread(args):
         config['GMAPS_KEY'] = args.gmaps_key
     else:
         config['GMAPS_KEY'] = load_credentials(os.path.dirname(os.path.realpath(__file__)))['gmaps_key']
-    app.run(threaded=True, debug=args.debug, host=args.host, port=args.port)
+    if args.no_server:
+        while True:
+            time.sleep(1000)
+    else:
+        app.run(threaded=True, debug=args.debug, host=args.host, port=args.port)
```
Comment (devinsba) [inline runserver.py]: I think the solution posted by @invisiblek is really how you want to do this

https://gist.github.com/invisiblek/e6f6cec123ec4431ee48475c48d20aa6

**Comment by JZ6** [comment]
@invisiblek could u do a pr with that

**Comment by LTHeaven** [comment]
@kingvibs @invisiblek today is the first time writing python for me and using git so you guys can take it on from here. im going to bed soon^^

**Comment by MonkeyLeeT** [comment]
@invisiblek We'll be ready to take your changes.

@LTHeaven Thanks for understanding bud :D

**Comment by LTHeaven** [comment]
no problem. i just saw this missing and implemented it JZ6 then said i should commit it

**Pull request closed without merging**